ui/apps/dev-server-ui/src/app/(dashboard)/functions/page.tsx

```diff
@@ -117,9 +118,21 @@ export default function FunctionList() {
       desc: false,
     },
   ]);
-  const [searchInput, setSearchInput] = useState('');
-  const [globalFilter, setGlobalFilter] = useState('');
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+  const { replace } = useRouter();
+  const query = searchParams.get('query');
+  const [searchInput, setSearchInput] = useState(query || '');
+  const [globalFilter, setGlobalFilter] = useState(query || '');
+
   const debouncedSearch = useDebounce(() => {
+    const params = new URLSearchParams(searchParams);
```

Comment:
We have custom hooks to simplify the search params: https://github.com/inngest/inngest/blob/main/ui/packages/components/src/hooks/useSearchParam.ts

Reply:
done.